### ISSUE TITLE
Accessibility and quality issues found in automatic tests for designsystemet.no

### ISSUE THREAD

**Issue by Camulos** [digdir/designsystemet#2501]
### Description of the bug

Automatic testing of designsystemet.no, tools used:
- Siteimprove
- Sortsite

Findings in categories:
- Accessibility
- Errors
- Usability
- Standards / Web
- Compatibility

Accessibility findings in WCAG categories A and AA should be prioritized.

Images from Sortsite (20. september 2024)
<img width="1034" alt="Skjermbilde 2024-09-20 kl  10 09 12" src="https://github.com/user-attachments/assets/d9fc1879-1137-40fd-849d-a06265e8a566">
<img width="1025" alt="Skjermbilde 2024-09-20 kl  10 08 40" src="https://github.com/user-attachments/assets/3d978b9a-e21c-476a-af45-2a2905763690">
<img width="869" alt="Skjermbilde 2024-09-20 kl  10 06 27" src="https://github.com/user-attachments/assets/95ef0110-c99c-448f-bb0c-414d82330b41">

### Steps To Reproduce

Scan using automatic tools.

**Comment by Camulos** [comment]
Level A finding in Sortsite: "A video plays longer than 5 seconds, without a way to pause it."
Relates to WCAG [2.2.2 Pause, stop, hide](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-pause.html)
Found 2 times, animations on front-page
![Skjermbilde 2024-09-24 kl  12 24 24](https://github.com/user-attachments/assets/a1410bcd-94c0-46ba-bd1e-0637171e68f2)
Respects user setting for reduced animations? Yes!
As the animations is looped, and the content is not related to real-time updates/events it may be necessary to also give the user means to stop the animation by interacting with it. However as it respects the user preferences it could be that this issue is not critical. (For discussion)

**Comment by Camulos** [comment]
Level A finding in Sortsite: "Document title must not be blank."
Relates to WCAG success criterion 2.4.2 Page titled.
Found 3 times in PDFs uploaded to the site:
![Skjermbilde 2024-09-24 kl  12 38 15](https://github.com/user-attachments/assets/c693ba12-336f-4720-ab7e-af3c72af6dd4)
The first two documents is generated from Confluence using the save as PDF feature, the documents should have been adjusted in Acrobat Pro before uploading here. 
The third document seem to be related to "Dødsfall og arv", based on the link-text.

The PDFs is used in this page: https://v0.designsystemet.no/grunnleggende/introduksjon/designprinsipper (found from the footer urls on the frontpage)

Adding title is fairly simple for PDFs, but if this is something we want to do, there may be other issues as well that the accessibility scan in Acrobat would find.

**Comment by Camulos** [comment]
Level A finding in Sortsite:
"SVG elements with graphic role attributes must have an accessible name." (SVG elements with role=img, role=graphics-document or role=graphics-symbol need an accessible name provided by the SVG title element or an ARIA label.)
Relates to WCAG success criterion: 1.1.1 Non-text content

Found on 28 pages, multiple times on some pages
<img width="1053" alt="Skjermbilde 2024-09-25 kl  08 22 27" src="https://github.com/user-attachments/assets/4e7c82dd-b7bb-42a0-b49c-aed711268e51">

It looks like this is a problem almost everywhere there is any SVG-files, for example the three icons used to illustrate the 3 main call to action cards from the front page.

Many of these SVGs could be hidden with aria hidden, but even if hidden they should most likely be given an labelled by and their first child should be an title?

**Comment by mimarz** [comment]
Moving this to an epic as we'll have different sub-tasks to fix this.

**Comment by Camulos** [comment]
Level A finding in Sortsite:
"Document title must not be blank."
Relates to WCAG success criterion 4.1.2 Name, Role, Value

Found 2 places and it looks like it is when we embed video from Vimeo:

- https://v0.designsystemet.no/bloggen/2023/derfor-trenger-vi-et-felles-designsystem
- https://v0.designsystemet.no/bloggen/2024/v1rc1

The frames/iframes need to have the title-attribute set to avoid SR read the filename (urls?) instead.

**Comment by Camulos** [comment]
Level A finding in Sortsite:
"Element li not allowed as child element in this context."
Relates to WCAG success criterion 1.3.1 Info and Relationships

Found on this page:
https://www.designsystemet.no/grunnleggende/designelementer/farger
In selections for brand or mode here:
<img width="544" alt="Skjermbilde 2024-09-25 kl  12 39 27" src="https://github.com/user-attachments/assets/d2feb043-fbda-45dd-bdc3-d0a99277f81c">


There are li-elements that are not set inside either ol or ul in these dropdowns.

**Comment by Camulos** [comment]
Level A finding in Sortsite:
"alt text should not contain placeholders like 'picture' or 'spacer'."
Relates to success criterion 1.1.1 Non-text Content

In this page: https://www.designsystemet.no/grunnleggende/designelementer/ikoner there is an image that uses the alt text "image" here:
<img width="583" alt="Skjermbilde 2024-09-25 kl  12 44 49" src="https://github.com/user-attachments/assets/e157ff65-ca77-44df-a93b-2d44f9d66204">

**Comment by Camulos** [comment]
Level AA finding in Sortsite:
"Phrases in a different language should be in a span or div with a lang attribute."
Relates to success criterion 3.1.2 Language of parts

Found in this page: https://www.designsystemet.no/grunnleggende/for-utviklere/komposisjon
Every time an code-example is inserted in the text, the SR will try to read them in Norwegian, however the codes should be read in english?

**Comment by Camulos** [comment]
Level AAA finding
"Several links on a page share the same link text, but go to different destinations."
Relates to WCAG success criterion 2.4.9 Link purpose (Link Only) **NB: AAA!**

Found on 4 pages:
https://www.designsystemet.no/bloggen/2024/v1rc1
https://www.designsystemet.no/monstre/feilmeldinger
https://www.designsystemet.no/monstre/systemvarsler
Link text: "Github" is used for 2 different urls in each of the pages above.

https://www.designsystemet.no/grunnleggende/introduksjon/designprinsipper
Link text "Design principles" is used for 3 different urls.

Amend by writing unique link texts for each url, reflecting the topic on the linked-to-page.

**Comment by mrosvik** [comment]
Closing this. @Febakke will check the reported issues again when testing, and create new issues if necessary.
